datarepo-clienttests/src/main/java/testscripts/BuildSnapshotWithFiles.java

```diff
@@ -0,0 +1,123 @@
+  List<Integer> batchSizes = new ArrayList<>();
+
+  public void setParameters(List<String> parameters) throws Exception {
+    if (parameters == null || parameters.size() < 3) {
+      throw new IllegalArgumentException(
+          "Required parameters: number of files, number of snapshots to create, batch size, batch size...");
+    } else {
+      filesToLoad = Integer.parseInt(parameters.get(0));
+      snapshotsToCreate = Integer.parseInt(parameters.get(1));
+      for (int i = 2; i < parameters.size(); i++) {
+        batchSizes.add(Integer.parseInt(parameters.get(i)));
+      }
+    }
+  }
+
+  // The purpose of this test is to measure the performance of building the snapshot file system
+  public void userJourney(TestUserSpecification testUser) throws Exception {
+    ApiClient apiClient = DataRepoUtils.getClientForTestUser(testUser, server);
+    RepositoryApi repositoryApi = new RepositoryApi(apiClient);
+
+    // set up and start bulk load job of small local files
+    BulkLoadArrayRequestModel arrayLoad =
+        BulkLoadUtils.buildBulkLoadFileRequest100B(filesToLoad, billingProfileModel.getId());
+    JobModel bulkLoadArrayJobResponse =
+        repositoryApi.bulkFileLoadArray(datasetSummaryModel.getId(), arrayLoad);
+
+    // wait for the job to complete
+    bulkLoadArrayJobResponse =
+        DataRepoUtils.waitForJobToFinish(repositoryApi, bulkLoadArrayJobResponse);
+    BulkLoadArrayResultModel result =
+        DataRepoUtils.expectJobSuccess(
+            repositoryApi, bulkLoadArrayJobResponse, BulkLoadArrayResultModel.class);
+    BulkLoadResultModel loadSummary = result.getLoadSummary();
+    assertThat(
+        "Number of successful files loaded should equal total files.",
+        loadSummary.getTotalFiles(),
+        equalTo(loadSummary.getSucceededFiles()));
+
+    // generate load for the simple datase
+    String testConfigGetIngestbucket = "jade-testdata";
+    String fileRefName =
+        "scratch/buildSnapshotWithFiles/" + FileUtils.randomizeName("input") + ".json";
+    IngestRequestModel ingestRequest =
+        BulkLoadUtils.makeIngestRequestFromLoadArray(
+            result, testConfigGetIngestbucket, fileRefName);
+
+    // load the data
+    JobModel ingestTabularDataJobResponse =
+        repositoryApi.ingestDataset(datasetSummaryModel.getId(), ingestRequest);
+
+    ingestTabularDataJobResponse =
+        DataRepoUtils.waitForJobToFinish(repositoryApi, ingestTabularDataJobResponse);
+    IngestResponseModel ingestResponse =
+        DataRepoUtils.expectJobSuccess(
+            repositoryApi, ingestTabularDataJobResponse, IngestResponseModel.class);
+    logger.info("Successfully loaded data into dataset: {}", ingestResponse.getDataset());
+
+    for (Integer batchSize : batchSizes) {
+      logger.info("Setting batch size to {}", batchSize);
+      DataRepoUtils.setConfigParameter(repositoryApi, "FIRESTORE_SNAPSHOT_BATCH_SIZE", batchSize);
+
+      // create the snapshots with all of the files
```

Comment:
Do we want only the snapshot creation to be part of the user journey? So that is the only part we optimize?

Reply:
Because i was looking at small pieces of the execution path, and not concerned about the total run time, it didn't matter to me what they user journey timing was. Now that i have moved the deletes out of the user journey, I think the full measurement will be what we it want to be.